### ISSUE TITLE
Maintainership and location of libstatistics_collector

### ISSUE THREAD

**Issue by christophebedard** [ros-tooling/libstatistics_collector#139]
This package is currently officially maintained by the tooling working group. The working group is going through some changes; see the 2022-05-20 meeting notes: https://docs.google.com/document/d/1UXWaEkT6ji0Z0uTtJJSni5Y6fiopM9qOOzv4ldeNaAM/edit#heading=h.gfl5k1w2acrq

@clalancette could Open Robotics take over ownership of this package, at least for the short term? Long term, I think it was decided in the middleware working group that the functionality here should be moved into rcl. However, it's not clear (currently) who would be in charge of doing that.

cc @emersonknapp

**Comment by christophebedard** [comment]
from [ROS 2 TSC Meeting Minutes for 5-19-2022](https://discourse.ros.org/t/ros-2-tsc-meeting-minutes-for-5-19-2022/25864):

> * [Biggs] Client libraries
>    * ...
>    * We were asked to move libstatistics_collector into the ros2 GitHub organisation because the tooling WG doesn’t want to maintain a core dependency of rclcpp.
>          * The decision made was to not move it, but instead to have the functionality of that library (it’s not a huge library) available in rcl, so that all client libraries can benefit.
>          * We would like a PR from AWS that does this.

Not sure who at AWS is going to do that though.

**Comment by clalancette** [comment]
TL;DR - If we have to, Open Robotics will reluctantly take over maintainership of the package, since it is a core dependency.  But we aren't on a sustainable path.

Essentially Open Robotics is a small team, and we already formally maintain hundreds of packages (and informally maintain more).  Every package we add to that causes our attention to be further divided, and thus the time we can spend on any individual package goes down.

On the whole, we prefer that packages be maintained by community members wherever possible, especially ones that were contributed by the community.  In this particular case, since this is a dependency of `rclcpp`, we'll take over maintainership if no one else steps up to do it, mostly because we have to.  But we'd really, really prefer if the tooling working group could find someone who would maintain it long-term.

**Comment by MichaelOrlov** [comment]
@clalancette @gbiggs I am trying to reiterate on this task.
Since `libstatistics_collector` package is a direct dependencies for `rclcpp`  and written purely on C++ it will make more sense to move it's functionality in `rclcpp` rather than to the `rcl` package.

To be able to move `libstatistics_collector` functionality in `rcl` package will need to rewrite it on `C`. It's gonna be a much more effort and doesn't looks reasonable at least for first iteration.

@cbuscaron Who from AWS could take care of migration `libstatistics_collector` package in to the `rclcpp` or `rcl` package?

**Comment by clalancette** [comment]
> Since `libstatistics_collector` package is a direct dependencies for `rclcpp` and written purely on C++ it will make more sense to move it's functionality in `rclcpp` rather than to the `rcl` package.

The thing is, there really isn't much reason to do that move.  If we are just going to maintain the current C++ code, then it may as well live in it's own repository.  The main motivation to move it to rcl was to make it available to other client libraries (like rclpy).

In short, I would either be in favor of leaving it where it is (in C++), or rewriting it in C and moving it to rcl.

**Comment by MichaelOrlov** [comment]
@clalancette Since no response from @cbuscaron  and no one else exposed willingness to rewrite `libstatistics_collector` on `C` for migrating it to the RCL package. On last Tooling WG meeting we decided to keep it as is and support in the scope of the current Tooling WG.  At the same time we decided to rename **"Tooling WG"** to **"Rosbag2 and tooling WG"** to emphasize that this working group mostly focused on `rosbag2` and the rest tooling staff is as addition to it.

Side note. I actually took a look on the source code and it looks like pretty simple and straight forward implementation written on C++ which I think I will be able to support in case if any issues will arise. I hope not many of them will arise though :smiley_cat: 

cc: @wjwwood

**Comment by cbuscaron** [comment]
Thank you @MichaelOrlov
